crates/typst/src/layout/flow.rs

```diff
@@ -875,3 +1017,35 @@ fn collect_footnotes(notes: &mut Vec<Packed<FootnoteElem>>, frame: &Frame) {
         }
     }
 }
+
+/// Collect all numbered paragraph lines in the frame.
+/// The 'prev_y' parameter starts at 0 on the first call to 'collect_par_lines'.
+/// On each subframe we encounter, we add that subframe's position to 'prev_y',
+/// until we reach a line's tag, at which point we add the tag's position
+/// and finish. That gives us the relative height of the line from the start of
+/// the initial frame.
+fn collect_par_lines(
+    lines: &mut Vec<(Abs, Packed<ParLineMarker>)>,
+    frame: &Frame,
+    prev_y: Abs,
+) {
+    for (pos, item) in frame.items() {
+        match item {
+            FrameItem::Group(group) => {
+                collect_par_lines(lines, &group.frame, prev_y + pos.y)
+            }
+            FrameItem::Tag(tag)
+                if !lines
+                    .iter()
+                    .any(|(_, line)| line.location() == tag.elem.location()) =>
```

Comment:
will this happen in practice? It makes the thing $O(n^2)$ (footnotes have that problem as well, but possibly a bit less pronounced in practice) and since we deduplicate by Y anyway it's perhaps unnecessary.

Reply:
Yeah, removing this seems to be totally fine, especially with deduplication later on.